website/docs/tutorials/08-cairo-1-support/01-testing.md

```diff
@@ -0,0 +1,78 @@
+---
+sidebar_label: Testing
+---
+
+# Running tests with Cairo 1
+
+:::info
+This functionality is in alpha stage, expect rapid iteration
+:::
+
+
+## Writing your first test
+
+To make a test function, you need to mark the function with a decorator `#[test]`
+
+```
+#[test]
+fn test_bool_operators() {
+    assert(1 == 1);
+}
+```
+
+To write a test which fails, you will need to use `panic`, here's how you do it:
+
+```
+use array::ArrayTrait;
+
+// Single value in panic payload
+#[test]
+fn test_panic_single_value() {
+    let mut data = array_new::<felt>();
+    array_append::<felt>(ref data, 21);
+    panic(data)
+}
+
+// Multiple values in panic payload
+#[test]
+fn test_panic_multiple_values() {
+    let mut data = array_new::<felt>();
+    array_append::<felt>(ref data, 101);
+    array_append::<felt>(ref data, 102);
+    array_append::<felt>(ref data, 103);
+    panic(data)
+}
+```
+
+## Running the tests
+
+To run cairo 1 tests, there is a special command called `test-cairo1`.
+It is a sister-command to `test` command, it will collect all the tests in the given directory/module, run them, and print out a summary.
+
+Tests are ran on Cairo VM, so no Starknet syscalls are available from the test code.
+
+See [command reference](../../cli-reference.md#test-cairo1) for more details on usage.
+
+:::info
+There is no support currently for starknet contracts at the moment, it's a work in progress.
+:::
+
+## Caveats
+### 1. Test collecting
+`test-cairo1` will collect all tests ending with `.cairo`, since there's no distinction between cairo 0 and cairo 1 files in terms of extension right now.
```

Comment:
```suggestion
`test-cairo1` will collect all tests ending with `.cairo` since there's no distinction between cairo 0 and cairo 1 files in terms of extension right now.
```